ynr/apps/candidates/tests/test_models.py

```diff
@@ -104,3 +151,85 @@ def test_has_results_and_no_results(self):
         all_winners = set(has_results + has_winner_no_result)
         self.assertEqual(set(Ballot.objects.has_results()), all_winners)
         self.assertEqual(set(Ballot.objects.no_results()), set(no_results))
+
+    def test_uncontested_when_winner_count_same_as_memberships(self):
+        parties = self.create_parties(2)
+        memberships = self.create_memberships(self.ballot, parties)
+        self.assertTrue(self.ballot.uncontested)
+
+    def test_uncontested_when_winner_count_is_less_than_memberships(self):
+        parties = self.create_parties(3)
+        memberships = self.create_memberships(self.ballot, parties)
+        self.assertFalse(self.ballot.uncontested)
+
+    def test_uncontested_when_winner_count_is_more_than_memberships(self):
+        parties = self.create_parties(1)
+        memberships = self.create_memberships(self.ballot, parties)
+        self.assertFalse(self.ballot.uncontested)
+
+
+class TestBallotQuerySetMethods(TestCase, SingleBallotStatesMixin):
```

Comment:
This is a duplicate of the `TestBallotQuerySetMethods` class above? I think the tests should be consolidated to a single class